client/landing/gutenboarding/onboarding-block/create-site/index.tsx

```diff
@@ -65,6 +70,9 @@ const CreateSite: React.FunctionComponent = () => {
 		return () => clearTimeout( id );
 	}, [] );
 
+	if ( ! hasRequiredFields ) {
+		return <Redirect to={ makePath( Step.IntentGathering ) } />;
```

Comment:
@roo2 I just realised we've got a bit of overlap in our PRs here. What do you think about moving this redirect to `/gutenboarding/onboarding-block/edit.tsx` where we have the other step-based redirect checks so that they're all in one place? In https://github.com/Automattic/wp-calypso/pull/41782/ I've added a few small functions for checking whether or not a step should be able to be accessed, so thought it might be a good place to group them together?

Your PR's further along than my one, though, so I'm happy to work around your changes :)

Reply:
hmm yes it looks like once your change is added, we won't need this line at all and the `canUseCreateSite` logic you're adding will handle it. I'd be happy to take this out of this PR and the effect would be that the user might get stuck on the `create-site` page if they logged in with a different browser until your PR is merged.

Reply:
Thanks @roo2! Just tested again after your changes, and if I log in in a different browser context the `?new` param ensures that a site gets created anyway, but defaults to twenty twenty theme and of course doesn't have any of the other info from signup. So, if it isn't too much trouble for you, how about we merge this one after #41782 is ready, once the  redirects are in place? That way the fallback should be handled already.

Reply:
oh that's not good. yes I'm happy to hold off on this until after #41782